bentoml/adapters/file_input.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2019 Atalaya Tech, Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import argparse
+import base64
+import contextlib
+from io import BytesIO
+from typing import Iterable
+
+from werkzeug.wrappers import Request
+
+from bentoml.exceptions import BadInput
+from bentoml.adapters.base_input import BaseInputAdapter
+from bentoml.marshal.utils import SimpleResponse, SimpleRequest
+
+
+class FileInput(BaseInputAdapter):
+    """Transform incoming file data from http request, cli or lambda event into numpy
+    array.
+
+    Handle incoming file data from different sources, transform them into numpy array
+    and pass down to user defined API functions
+
+    Args:
```

Comment:
@bojiang could you add some more usage examples here? including a sample BentoService class using this adapter and sample curl request and CLI command invoking the API using this input adapter, similar to the one we had in multi_image_adapter here: https://github.com/bentoml/BentoML/blob/master/bentoml/adapters/multi_image_input.py#L54